pkg/loop/internal/chain_reader.go

```diff
@@ -106,19 +115,21 @@ type chainReaderServer struct {
 }
 
 func (c *chainReaderServer) GetLatestValue(ctx context.Context, request *pb.GetLatestValueRequest) (*pb.GetLatestValueReply, error) {
-	var bc types.BoundContract
-	bc.Name = request.Bc.Name[:]
-	bc.Address = request.Bc.Address[:]
-	bc.Pending = request.Bc.Pending
-
-	params := &map[string]any{}
+	contractName := request.ContractName
+	params, err := getContractEncodedType(contractName, request.Method, c.impl, true)
+	if err != nil {
+		return nil, err
+	}
 
-	if err := decodeVersionedBytes(params, request.Params); err != nil {
+	if err = decodeVersionedBytes(params, request.Params); err != nil {
 		return nil, err
 	}
 
-	retVal := &map[string]any{}
-	err := c.impl.GetLatestValue(ctx, bc, request.Method, params, retVal)
+	retVal, err := getContractEncodedType(contractName, request.Method, c.impl, false)
+	if err != nil {
+		return nil, err
+	}
+	err = c.impl.GetLatestValue(ctx, contractName, request.Method, params, retVal)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
I thought we discussed this but I'm not sure: Why do we respond with the `CurrentEncodingVersion` below on 126 instead of the version that they used to initiate this call, `request.Params.Version`?

Reply:
I'm not sure, I didn't change that from what was there before.  It makes sense to use the params version.  Should we have the Decoding ask for a version that you also want back?  It's currently sent raw bytes so it doesn't need an encoding for the object in the request.

Reply:
Done, and added a variable in `Codec` since it doesn't receive `VersionedBytes`